### PR TITLE
Possibility to override cloud endpoint

### LINKED ISSUE #23: Possibility to override cloud endpoint
the CLI currently always uses the hosted instance of Monokle Cloud for authenticating keys and retrieving policies - we will need to be able to override the endpoint for use with enterprise deployments of Cloud

### PR DESCRIPTION
**Pull request by f1ames**
This PR fixes #23. it introduced mechanics to set origin according to initial issue, https://github.com/kubeshop/monokle-cli/issues/23#issuecomment-1808310030 and https://github.com/kubeshop/monokle-cli/issues/23#issuecomment-1814178247:

### `login`

For `login` command one could use `--origin` flag (highest priority) and also `MONOKLE_ORIGIN` env var (lower priority) will be read and used if set. If neither is provided, user will be prompted if he/she wants to use custom origin. Origin is preserved upon login so all commands (as long as you stay logged in) will connect to this origin.

### `validate` and `config`

If user is logged in, origin set during login process is used. Still, no matter if authenticated or not, one can use `--apiToken` + `--project` + `--origin` to run those commands with different origin (or just `--apiToken` + `--project` to run on the same origin but with different credentials).

At the beginning I thought to take `MONOKLE_ORIGIN` env into account also for `validate` and `config` command runs but it really seemed like it will make it more confusing and complicated (both for the user and from the code perspective).

monokle 20231120-1

---

Requires https://github.com/kubeshop/monokle-core/pull/567.

## Changes

- As above.

## Fixes

- As above.

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [x] updated the docs
- [ ] added a test

### CODE REVIEW

File: src/commands/login.io.ts [hunk 1]
```diff
@@ -1,6 +1,28 @@
 import prompts from "prompts";
 import { C } from "../utils/screens.js";
 
+export const promptForOrigin = async () => {
+  const ownOrigin = await prompts({
+    type: 'confirm',
+    name: 'value',
+    message: 'Would you like the use custom origin?',
```
Comment (WitoDelnat) [inline src/commands/login.io.ts]: I would guide the users a bit more here similar to GitHub CLI. Providing a custom origin is a means to connect with Enterprise.

```bash
 ▶ gh auth login
? What account do you want to log into?  [Use arrows to move, type to filter]
> GitHub.com
  GitHub Enterprise Server
```

File: src/commands/login.io.ts [hunk 2]
```diff
@@ -1,6 +1,28 @@
...
+  });
+
+  return Boolean(ownOrigin.value);
+};
+
+export const promptForOriginValue = async () => {
+  const originInput = await prompts({
+    type: 'text',
+    name: 'value',
+    message: 'Enter Monokle remote web app instance URL',
```
Comment (WitoDelnat) [inline src/commands/login.io.ts]: Would plain hostname also work? e.g.`monokle.example.com`. This makes it easier for the end users by not having to type the URL scheme. (We can still normalise it to an origin for internal usage afterwards)

`Enter Monokle Enterprise hostname:`

**Review by WitoDelnat — COMMENTED** [review]
LGTM. Just a minor remark about the prompts.

I do agree that MONOKLE_ORIGIN might have some odd interactions when you are logged in to origin_a and then change MONOKLE_ORIGIN to origin_b. I guess its fine for now though so let's keep the current interaction going!

File: src/commands/login.io.ts (hunk 2 above)
Comment (f1ames) [reply src/commands/login.io.ts]: Yes, you can use hostname without protocol like `app.staging.monokle.com`. It's normalized by synchronizer.

File: src/commands/login.io.ts (hunk 1 above)
Comment (f1ames) [reply src/commands/login.io.ts]: Looks better now indeed:

monokle 20231120-2

Fixed in https://github.com/kubeshop/monokle-cli/pull/39/commits/78dd716a705460018abdcf53d74d1517d3b3d734.

**Review by WitoDelnat — APPROVED** [review]

**Pull request merged**